scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContext.java

```diff
@@ -73,23 +73,24 @@
 
 
     public TaskContext(ExecutableContainer executableContainer, TaskLauncherInitializer initializer,
-            TaskResult[] previousTasksResults, String scratchURI, String inputURI, String outputURI,
-            String userURI,
-            String globalURI, String progressFilePath, String currentNodeHostname) throws NodeException {
-        this(executableContainer, initializer, previousTasksResults, scratchURI, inputURI, outputURI, userURI,
+                       TaskResult[] previousTasksResults, String scratchURI, String cacheURI, String inputURI, String outputURI,
+                       String userURI,
+                       String globalURI, String progressFilePath, String currentNodeHostname) throws NodeException {
+        this(executableContainer, initializer, previousTasksResults, scratchURI, cacheURI, inputURI, outputURI, userURI,
                 globalURI, progressFilePath, currentNodeHostname, null);
     }
```

Comment:
We should create a Builder for this class there is too many arguments for the constructors. It is not maintainable.